gtsfm/loader/hilti_loader.py

```diff
@@ -273,22 +271,8 @@ def map_index_to_camera(self, index: int) -> int:
     def map_image_idx_to_rig(self, index: int) -> int:
         return index // NUM_CAMS
 
-    def create_computation_graph_for_relative_pose_priors(
-        self, pairs: List[Tuple[int, int]]
-    ) -> Dict[Tuple[int, int], Delayed]:
-        # Hack: just add all possible pairs which belong to the same rig (as it will have hard relative prior)
-        pairs = set(pairs)
-        for i in range(len(self)):
-            for j in range(i + 1, i + NUM_CAMS - 1):
-                if self.map_image_idx_to_rig(i) == self.map_image_idx_to_rig(j):
-                    pairs.add((i, j))
-                else:
-                    break
-
-        return {(i1, i2): dask.delayed(self.get_relative_pose_prior)(i1, i2) for i1, i2 in pairs}
-
-    def get_all_relative_pose_priors(self) -> Dict[Tuple[int, int], Optional[PosePrior]]:
-        pairs = set(self.get_valid_pairs())
+    def get_relative_pose_priors(self, pairs: List[Tuple[int, int]]) -> Dict[Tuple[int, int], Optional[PosePrior]]:
+        pairs = set(self.get_valid_pairs())  # overwites argument !
```

Comment:
We should not do this. Instead have `pairs = set(pairs)`. In the future, the `is_valid_pair` and `get_valid_pairs` will be killed from the loader and we will have retriever taking care of providing pairs.

Reply:
Done